torchvision/datasets/mnist.py

```diff
@@ -440,19 +450,20 @@ def open_maybe_compressed_file(path):
     return open(path, 'rb')
 
 
-def read_sn3_pascalvincent_tensor(path, strict=True):
+SN3_PASCALVINCENT_TYPEMAP = {
+    8: (torch.uint8, np.uint8, np.uint8),
+    9: (torch.int8, np.int8, np.int8),
+    11: (torch.int16, np.dtype('>i2'), 'i2'),
+    12: (torch.int32, np.dtype('>i4'), 'i4'),
+    13: (torch.float32, np.dtype('>f4'), 'f4'),
+    14: (torch.float64, np.dtype('>f8'), 'f8')
+}
+
+
+def read_sn3_pascalvincent_tensor(path: Union[str, IO], strict: bool = True) -> torch.Tensor:
     """Read a SN3 file in "Pascal Vincent" format (Lush file 'libidx/idx-io.lsh').
        Argument may be a filename, compressed filename, or file object.
     """
-    # typemap
-    if not hasattr(read_sn3_pascalvincent_tensor, 'typemap'):
```

Comment:
Not sure what the original intention was to assign this dict to the function in the first run instead as a constant on the outside. I didn't manage to annotate it so that `mypy` didn't complain, so I moved it outside.

Reply:
It's a micro-optimization, but It's ok to change it as you did. I would even put it inside the function itself, so that it doesn't appear in the global namespace of `mnist`.